docs/book/ceph-authorization.md

```diff
@@ -0,0 +1,1197 @@
+    rgw opa url = opa.rook-ceph:8181/v1/data/ceph/authz/allow
+```
+
+### 6. Create the Ceph Object Store
+
+Save the object store spec as **object.yaml**:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStore
+metadata:
+  name: my-store
+  namespace: rook-ceph
+spec:
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 1
+  dataPool:
+    failureDomain: osd
+    replicated:
+      size: 1
+  gateway:
+    type: s3
+    sslCertificateRef:
+    port: 80
+    securePort:
+    instances: 1
+    allNodes: false
+    placement:
+    resources:
+```
+
+```bash
+kubectl create -f object.yaml
+```
+
+When the object store is created, the RGW service with the S3 API will be started in the cluster. The Rook operator will create all the pools and other resources necessary to start the service.
+
+Check that the RGW pod is `Running`.
+
+```bash
+kubectl -n rook-ceph get pod -l app=rook-ceph-rgw
+```
+
+Rook sets up the object storage so pods will have access internal to the cluster. Create a new service for external access. We will need the external RGW service for exercising our OPA policy.
+
+Save the external service as **rgw-external.yaml**:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: rook-ceph-rgw-my-store-external
+  namespace: rook-ceph
+  labels:
+    app: rook-ceph-rgw
+    rook_cluster: rook-ceph
+    rook_object_store: my-store
+spec:
+  ports:
+  - name: rgw
+    port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: rook-ceph-rgw
+    rook_cluster: rook-ceph
+    rook_object_store: my-store
+  sessionAffinity: None
+  type: NodePort
+```
+
+Create the external service.
+
+```bash
+kubectl create -f rgw-external.yaml
+```
+
+Check that both the `internal` and `external` RGW services are `Running`.
+
+```bash
+kubectl -n rook-ceph get service rook-ceph-rgw-my-store rook-ceph-rgw-my-store-external
+```
+
+### 7. Create Object Store Users
+
+Create two object store users `Alice` and `Bob`.
+
+**object-user-alice.yaml**
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: alice
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "Alice"
+```
+
+**object-user-bob.yaml**
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: bob
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "Bob"
+```
+
+Now create the users.
+
+```bash
+kubectl create -f object-user-alice.yaml
+kubectl create -f object-user-bob.yaml
+```
+
+When the object store user is created the Rook operator will create the RGW user on the object store `my-store`, and store the user's Access Key and Secret Key in a Kubernetes secret in the namespace `rook-ceph`.
+
+### 8. Understanding the OPA policy
+
+As we saw earlier, the OPA spec contained a ConfigMap that defined the policy to be used to authorize requests received by the `Ceph Object Gateway`. Below is the policy:
+
+**authz.rego**
+
+{%ace lang='python'%}
```

Comment:
This looks odd [when viewed on github](https://github.com/open-policy-agent/opa/blob/59d2a64be52f4c8d88f1ebcb48a37145df4ab601/docs/book/ceph-authorization.md#8-understanding-the-opa-policy)... But I guess it's fine when rendered with gitbook...